src/image_transfer.py

```diff
@@ -0,0 +1,286 @@
+            # is present in the list and return True if it is present.
+            LOGGER.info(f"[Override Enabled] Transfer override enabled for the image {image_uri}")
+            return True
+    return False
+
+
+def transfer_image(autopatch_image_repo, autopatch_image_tag_list, beta_repo):
+    """
+    Transfers the image from the AutoPatch ECR to the Beta ECR. This method iterates through all the tags attached
+    in the AutoPatch image and removes "-autopatch" from those tags to form the appropriate beta_tag for the beta repository.
+    It then tags the image with {beta_repo}:{beta_tag} and pushes the image. It also attaches the autopatch tags to the beta image.
+    In other words, it tags the image wit {beta_repo}:{autopatch_tag} and pushes the image to ensure that autopatch tags are
+    preserved with the image as well. It DOES NOT transfer those autopatch_tags to the beta repo that have "benchmark-tested" in it.
+
+    :param autopatch_image_repo: str, Image URI of the AutoPatch Image.
+    :param autopatch_image_tag_list: list, List of all the image tags that were attached to the image in autopatch ECR
+    :param beta_repo: str, Beta Repo
+    """
+    for autopatch_tag in autopatch_image_tag_list:
+        beta_tag = autopatch_tag.replace("-autopatch", "")
+        run(f"docker tag {autopatch_image_repo}:{autopatch_tag} {beta_repo}:{beta_tag}", hide=True)
+        LOGGER.info(f"docker push {beta_repo}:{beta_tag}")
+        run(f"docker push {beta_repo}:{beta_tag}", hide=True)
+        if "benchmark-tested" not in autopatch_tag:
+            run(
+                f"docker tag {autopatch_image_repo}:{autopatch_tag} {beta_repo}:{autopatch_tag}",
+                hide=True,
+            )
+            LOGGER.info(f"docker push {beta_repo}:{autopatch_tag}")
+            run(f"docker push {beta_repo}:{autopatch_tag}", hide=True)
+
+
+def is_image_transferable(autopatch_image_uri, beta_image_uri, image_transfer_override_flags):
+    """
+    Checks if an image is transferable or not. It first checks if the image_transfer_flag has been enabled. In case
+    yes, it trasfers the image. In case not, it checks if the latest benchmark image in the Beta ECR is stale for more
+    than 5 days. If yes, conducts the transfer. If not, does not conduct the transfer.
+
+    :param autopatch_image_uri: str, Image URI of the AutoPatch image
+    :param beta_image_uri: str, Image URI of the Beta image
+    :param image_transfer_override_flags: dict, Image Override flags derived from the s3 Bucket
+    :return: boolean, True if the transfer can happen, False otherwise.
+    """
+    if is_image_transfer_enabled_by_override_flags(
+        image_uri=autopatch_image_uri, image_transfer_override_flags=image_transfer_override_flags
+    ):
+        return True
+    beta_image_push_time = get_push_time_of_image_from_ecr(image_uri=beta_image_uri)
```

Comment:
If this script already imports `test.test_utils`, then why does it need the use of a wrapper function around `get_image_push_time_from_ecr` in test_utils?

Reply:
That's a nice suggestion. I have removed the wrapper function altogether. Thanks!